docs/proposals/cli.md

```diff
@@ -0,0 +1,328 @@
+| --page    | -p        |                                                                         | false              | Specifies whether to display one page of text at a time or all text at one time.                                                                                 |
+| --rows    | -r        | integer                                                                 | 24                 | Specifies the number of rows per page to display when the **-p** parameter is on. You can specify a value in the range 1 - 100.                                  |
+| --header  | -h        |                                                                         | true               | Specifies whether to display the table header. Use the default value of on to display the table header. Use off to hide the table header.                        |
+
+**Example:**
+
+```bash
+kudoctl get instances
+```
+
+</details>
+
+---
+
+<details> <summary><h3>get instance &lt;id&gt;</h3></summary>
+
+Get details about the instance.
+
+**Arguments** :
+
+`id` : the id of the instance
+
+**Flags :**  
+
+| Name     | Shorthand | Values                                       | Default            | Description                                                            |
+| -------- | --------- | -------------------------------------------- | ------------------ | ---------------------------------------------------------------------- |
+| --format |           | `"json"`,`"human_readable"`,`"yaml"`,`"xml"` | `"human_readable"` | The output format of the resource definition,  yml is the same as yaml |
+| --help   | -h        |                                              | false              | show help of the function.                                             |
+
+**Example:**
+
+```bash
+kudoctl get instance id9878
+```
+
+</details>
+  
+---
+
+<details> <summary><h3>delete resource &lt;id&gt;</h3></summary>
+
+Delete a resource definition and all the instances of this resource. On success the command outputs no information.
+
+**Arguments** :
+
+`id` : the id of the resource
+
+**Flags :**  
+
+| Name   | Shorthand | Values | Default | Description                |
+| ------ | --------- | ------ | ------- | -------------------------- |
+| --help | -h        |        | false   | show help of the function. |
+
+**Example:**
+
+```bash
+kudoctl delete resource id8989
+```
+
+</details>
+
+---
+
+<details> <summary><h3>delete instance &lt;id&gt;</h3></summary>
+
+Delete and stop an instance. On success the command outputs no information.
+
+**Arguments** :
+
+`id` : the id of the instance.
+
+**Example:**
+
+```bash
+kudoctl delete instance id9898
+```
+  
+</details>
+
+---
+
+<details> <summary><h3>create &lt;kind&gt;</h3></summary>
+
+Create a resource definition. By default if a resource with the same name exists, the resource will be updated, add the `--no-update` flag if you don’t want this behavior.
```

Comment:
If you want to support imperative mode and update, you should add an `update`command and avoid piggy backing the `create`one.